sherpa/fit.py

```diff
@@ -1430,8 +1461,9 @@ def fit(self,
                         param_warnings += f"WARNING: parameter value {par.fullname} is at its minimum boundary {par.min}\n"
                     if sao_fcmp(par.val, par.max, tol) == 0:
                         param_warnings += f"WARNING: parameter value {par.fullname} is at its maximum boundary {par.max}\n"
+            record_steps = cb.record_steps
 
-        output = (status, newpars, fval_new, msg, imap)
+        output = (status, newpars, fval_new, msg, imap, record_steps)
```

Comment:
The trouble here is that `FitResults` accepts an `OptReturn` type for its second argument. I am making use of this type in my clean-up code (as I've found having useful type statements added to sherpa.optmethods/estmethods./fit quite helpful in identifying how information flows), and this extra information isn't needed/wanted in the optimizer code.

So I suggest instead that we send the `record_steps`  in some other way to `FitResults`.

**Or** we can take advantage of `OptReturn` having a get-out-of-jail-free type last argument - in this case `imap`, a dictionary, and so do something like

````python
imap[some-key-or-other] = record_steps
output = (status, newpars, fval_new, msg, imap)
```

and then `FitResults` will fish the value out of this dictionary.